plugins/traction_innkeeper/traction_innkeeper/v1_0/tenant/routes.py

```diff
@@ -131,7 +133,19 @@ async def tenant_wallet_update(request: web.BaseRequest):
         settings["default_label"] = label
     if image_url is not None:
         settings["image_url"] = image_url
-
+    if "ACAPY_ENDORSER_ROLE" in extra_settings:
+        if extra_settings.get("ACAPY_ENDORSER_ROLE") == "author":
+            settings["endorser.endorser"] = False
+        elif extra_settings.get("ACAPY_ENDORSER_ROLE") == "endorser":
+            settings["endorser.author"] = False
+            settings["endorser.auto_request"] = False
+            settings["endorser.auto_write"] = False
+        elif extra_settings.get("ACAPY_ENDORSER_ROLE") == "none":
+            settings["endorser.author"] = False
+            settings["endorser.endorser"] = False
+            settings["endorser.auto_request"] = False
+            settings["endorser.auto_write"] = False
```

Comment:
This logic I think should reflect in the tenantUI too. It was pretty confusing to me when some switches weren't updating (or updating on their own) based on the endorser role setting in the frontend 😄 
Happy to leave as a Tenant UI UX improvement for now an I create a ticket and fix quickly if we want. (I can probably to a better grouping of those settings if they are related as well then)

Reply:
Re `endorser role` not tracking, there was a bug in the initial implementation in ACA-Py. This [PR](https://github.com/hyperledger/aries-cloudagent-python/pull/2331) merged yesterday should fix this.

I don't think there is a need to include this logic in tenantUI, the above fix should resolve issues around endorser role switches not updating.

Reply:
Ah ok thanks! Didn't catch that there were changes in 0.9 around this